Source/WebCore/platform/graphics/displaylists/DisplayListItem.cpp

```diff
@@ -51,34 +52,55 @@ bool isValid(const Item& item)
 }
 
 template<class T>
-inline static std::optional<RenderingResourceIdentifier> applyFilteredImageBufferItem(GraphicsContext& context, const ResourceHeap& resourceHeap, const T& item)
+inline static std::optional<RenderingResourceIdentifier> applyFilteredImageBufferItem(GraphicsContext& context, const ResourceHeap& resourceHeap, const T& item, ApplyItemOptions options)
 {
     auto resourceIdentifier = item.sourceImageIdentifier();
     auto sourceImage = resourceIdentifier ? resourceHeap.getImageBuffer(*resourceIdentifier) : nullptr;
     if (UNLIKELY(!sourceImage && resourceIdentifier))
         return resourceIdentifier;
 
+#if USE(SKIA)
+    if (sourceImage && options == ApplyItemOptions::FlushImagesAndWaitForCompletion)
+        sourceImage->waitForAcceleratedRenderingFenceCompletion();
+#else
+    UNUSED_PARAM(options);
+#endif
```

Comment:
I think this code should be in `ResourceHeap::getImageBuffer()`. So it will not be repeated in `applyImageBufferItem()`.

Reply:
Good idea!